vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -939,4 +941,21 @@ private void fireSizeEvent() {
             fireEvent(new SizeChangeEvent<>(this, newSize));
         }
     }
+
+    @SuppressWarnings("unchecked")
+    private IdentifierProvider<T> getIdentifierProvider() {
+        IdentifierProvider<T> identifierProviderObject =
+                (IdentifierProvider<T>) ComponentUtil.getData(this,
+                        IdentifierProvider.class);
```

Comment:
Non blocking comment: this whole method `getIdentifiedProvider()` could probably be refactored to a reusable utility method `getIdentifierProvider(Component)` in `flow-data`.

But it can be done later and the part for `getDataProvider` should be solved then